adapters/wazirx-nft-Xauction-bsc/index.js

```diff
@@ -0,0 +1,176 @@
+        const timestamp = moment.unix(block.timestamp).utc();
+        let nativePrice;
+        let po = 0;
+        let seller = {
+            address: null,
+            price: 0,
+        };
+
+        const tokenId = event.returnValues._tokenId;
+
+        const transaction = await this.sdk.getTransactionReceipt(event.transactionHash);
+
+        const auctionContract = this.sdk.web3.eth.abi.encodeParameter("address", this.contract);
+        const logs = transaction.logs.filter(
+            tx => tx.topics[0] === TRANSFER_TOPIC && tx.topics[1].toLowerCase() === auctionContract,
+        );
+
+        const bnbusd = await this.getPrice(block.timestamp, this.currencyCode["2"].token_address);
+
+
+        if (logs.length) {
+            // Look for highest amount transfer
+            const price = logs.reduce((sum, item, index) => {
+                const decodedPrice = new BigNumber(item.data).dividedBy(10 ** 8).toNumber();
+                const address = this.sdk.web3.eth.abi.decodeParameter("address", item.topics[2]);
+                sum += decodedPrice;
+
+                if (index === 0) {
+                    seller = {
+                        address: address,
+                        price: decodedPrice,
+                    };
+                } else if (decodedPrice > seller.price) {
+                    seller = {
+                        address: address,
+                        price: decodedPrice,
+                    };
+                }
+                return sum;
+            }, 0);
+            po = await this.getPrice(block.timestamp, this.currencyCode["1"].token_address);
+            nativePrice = new BigNumber(price).times(po.price).dividedBy(bnbusd.price);
+        } else {
+            let events = [];
+            let _event = {};
+            let blockRange = event.blockNumber;
+            // Look for latest BidPlaced event by bid winner to find price
+            while (!events.length) {
+                events = await this.sdk.getPastEvents("BidPlaced", {
+                    filter: { _tokenId: tokenId, _from: event.returnValues._bidWinner, _currency: "2" },
+                    fromBlock: blockRange - 5000,
+                    toBlock: blockRange,
+                });
+                blockRange -= 5000;
+            }
+
+            _event = events[events.length - 1];
+            nativePrice = new BigNumber(_event.returnValues._amount).dividedBy(10 ** 18);
+            po = bnbusd;
+            seller = {
+                address: event.returnValues._settledBy,
+            };
+        }
+
+        const entity = {
+            provider_name: this.name,
+            provider_contract: this.contract,
+            protocol: AVAILABLE_PROTOCOLS.BSC,
+            nft_contract: this.nftContract,
+            nft_id: tokenId,
+            token: this.token,
+            token_symbol: this.token_symbol,
+            amount: 1,
+            price: nativePrice.toNumber() || 1,
```

Comment:
What is the purpose of fallback here? Statistics would be inaccurate.

Reply:
Sometimes the getPrice API fails to provide price details for the provided block timestamp.  This fallback is just to pass the test cases

Reply:
Do you have an example of API error? We would like to investigate what is the reason.

Reply:
I know how to reproduce, but API (http://nft-sales-service.dappradar.com ) responds with 401 Forbidden. any reasons?

Reply:
![Screenshot 2022-03-11 at 9 35 21 AM](https://user-images.githubusercontent.com/83953812/157800160-c6073dfe-c7c4-4368-abd9-909278c389d2.png)


I got an error for block 14865746 and timestamp 1643691130. But now it's not reproducible.